src/instructlab/sdg/llmblock.py

```diff
@@ -167,29 +195,33 @@ def __init__(
                     **self._load_config(config)
                 )
 
-    def _generate(self, samples, **gen_kwargs) -> str:
+    def _format_prompt(self, sample: Dict) -> str:
         if isinstance(self.prompt_template, dict):
-            prompts = [
-                self.model_prompt.format(
-                    prompt=self.prompt_template[sample[self.selector_column_name]]
-                    .format(**sample)
-                    .strip()
-                )
-                for sample in samples
-            ]
-        else:
-            prompts = [
-                self.model_prompt.format(
-                    prompt=self.prompt_template.format(**sample).strip()
-                )
-                for sample in samples
-            ]
-        response = self.client.completions.create(
-            prompt=prompts, **{**self.defaults, **gen_kwargs}
-        )
-        return [choice.text.strip() for choice in response.choices]
+            return (self.prompt_template[sample[self.selector_column_name]]
+                    .format(**sample).strip())
+
+        return self.prompt_template.format(**sample).strip()
 
     def validate(self, prompt_template: str, input_dict: Dict[str, Any]) -> bool:
         if isinstance(prompt_template, dict):
             prompt_template = prompt_template[input_dict[self.selector_column_name]]
         return super()._validate(prompt_template, input_dict)
+
+
+def server_supports_batched(client, model_id: str) -> bool:
+    supported = getattr(client, "server_supports_batched", None)
+    if supported is not None:
+        return supported
+    try:
+        # Make a test call to the server to determine whether it supports
+        # multiple input prompts per request and also the n parameter
+        response = client.completions.create(
+            model=model_id, prompt=["test1", "test2"], max_tokens=1, n=3
+        )
+        # Number outputs should be 2 * 3 = 6
+        supported = len(response.choices) == 6
+    except openai.InternalServerError:
+        supported = False
+    client.server_supports_batched = supported
```

Comment:
Hmm, I understand that you want to cache this ... but I don't like setting a new attribute on a class we don't own

I guess this could be removed with a move to `FlowParams`